targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,125 @@
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace PlayFab.UUnit
+{
+    public class PluginManagerTest : UUnitTestCase
+    {
+        private class CustomSerializerPlugin : ISerializerPlugin
+        {
+            public T DeserializeObject<T>(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public T DeserializeObject<T>(string serialized, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object DeserializeObject(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public string SerializeObject(object obj)
+            {
+                throw new NotImplementedException();
+            }
+
+            public string SerializeObject(object obj, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class CustomTransportPlugin : ITransportPlugin
+        {
+            public string Name;
+
+            public async Task<object> DoPost(string urlPath, object request, Dictionary<string, string> headers)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// CLIENT AND SERVER API
+        /// Test that plugin manager returns default plugins if they are not set.
+        /// </summary>
+        [UUnitTest]
+        public void PluginManagerDefaultPlugins(UUnitTestContext testContext)
+        {
+            var playFabSerializer = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
+            var playFabTransport = PluginManager.GetPlugin<ITransportPlugin>(PluginContract.PlayFab_Transport);
+
+            testContext.NotNull(playFabSerializer);
+            testContext.NotNull(playFabTransport);
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        /// <summary>
+        /// CLIENT AND SERVER API
+        /// Test that plugin manager can set and return a custom plugin.
+        /// </summary>
+        [UUnitTest]
+        public void PluginManagerCustomPlugin(UUnitTestContext testContext)
+        {
+            var playFabSerializer = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
+            var customSerializer = new CustomSerializerPlugin();
+            try
+            {
+                // Set a custom serializer plugin
+                PluginManager.SetPlugin(customSerializer, PluginContract.PlayFab_Serializer);
+
+                // Get serializer plugin from manager
+                var serializerPlugin = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
+
+                // Verify
+                testContext.True(object.ReferenceEquals(serializerPlugin, customSerializer));
```

Comment:
I like to use the phrasing "expected" and "actual" as much as possible during tests.
I need to read the above lines pretty carefully to decide which is which in this case.